oshmem/include/pshmem.h

```diff
@@ -1508,6 +2389,375 @@ OSHMEM_DECLSPEC void pshmem_longlong_inc(long long *target, int pe);
+OSHMEM_DECLSPEC void pshmem_ctx_uint64_atomic_fetch_nbi(shmem_ctx_t ctx, uint64_t *fetch, const uint64_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_ctx_size_atomic_fetch_nbi(shmem_ctx_t ctx, size_t *fetch, const size_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_ctx_ptrdiff_atomic_fetch_nbi(shmem_ctx_t ctx, ptrdiff_t *fetch, const ptrdiff_t *target, int pe);
+
+OSHMEM_DECLSPEC void pshmem_int_atomic_fetch_nbi(int *fetch, const int *target, int pe);
+OSHMEM_DECLSPEC void pshmem_long_atomic_fetch_nbi(long *fetch, const long *target, int pe);
+OSHMEM_DECLSPEC void pshmem_longlong_atomic_fetch_nbi(long long *fetch, const long long *target, int pe);
+OSHMEM_DECLSPEC void pshmem_uint_atomic_fetch_nbi(unsigned int *fetch, const unsigned int *target, int pe);
+OSHMEM_DECLSPEC void pshmem_ulong_atomic_fetch_nbi(unsigned long *fetch, const unsigned long *target, int pe);
+OSHMEM_DECLSPEC void pshmem_ulonglong_atomic_fetch_nbi(unsigned long long *fetch, const unsigned long long *target, int pe);
+OSHMEM_DECLSPEC void pshmem_float_atomic_fetch_nbi(float *fetch, const float *target, int pe);
+OSHMEM_DECLSPEC void pshmem_double_atomic_fetch_nbi(double *fetch, const double *target, int pe);
+OSHMEM_DECLSPEC void pshmem_int32_atomic_fetch_nbi(int32_t *fetch, const int32_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_int64_atomic_fetch_nbi(int64_t *fetch, const int64_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_uint32_atomic_fetch_nbi(uint32_t *fetch, const uint32_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_uint64_atomic_fetch_nbi(uint64_t *fetch, const uint64_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_size_atomic_fetch_nbi(size_t *fetch, const size_t *target, int pe);
+OSHMEM_DECLSPEC void pshmem_ptrdiff_atomic_fetch_nbi(ptrdiff_t *fetch, const ptrdiff_t *target, int pe);
+
+#if OSHMEM_HAVE_C11
+#define pshmem_atomic_fetch_nbi(...)                                        \
+    _Generic(&*(__OSHMEM_VAR_ARG1(__VA_ARGS__)),                            \
+            shmem_ctx_t:  _Generic((__OSHMEM_VAR_ARG2(__VA_ARGS__)),        \
+                int*:         pshmem_ctx_int_atomic_fetch_nbi,              \
+                long*:        pshmem_ctx_long_atomic_fetch_nbi,             \
+                long long*:   pshmem_ctx_longlong_atomic_fetch_nbi,         \
+                unsigned int*:       pshmem_ctx_uint_atomic_fetch_nbi,      \
+                unsigned long*:      pshmem_ctx_ulong_atomic_fetch_nbi,     \
+                unsigned long long*: pshmem_ctx_ulonglong_atomic_fetch_nbi, \
+                float*:       pshmem_ctx_float_atomic_fetch_nbi,            \
+                double*:      pshmem_ctx_double_atomic_fetch_nbi,           \
+                size_t*:      pshmem_ctx_size_atomic_fetch_nbi,             \
+                ptrdiff_t*:      pshmem_ctx_ptrdiff_atomic_fetch_nbi,       \
+                default:      __opshmem_datatype_ignore),                   \
```

Comment:
inconsistency: this section is missing in non-context section of macro (on lines 2450+)
as I can see size_t and ptrdiff_t are covered by generic datatypes

Reply:
Good catch. Adding them.